src/analyses/apron/relationAnalysis.apron.ml

```diff
@@ -365,7 +532,19 @@ struct
       |> List.filter RD.Tracked.varinfo_tracked
       |> List.map RV.local
     in
-    RD.remove_vars_with new_rel local_vars;
+    let pointsToList = match e with 
+      | Some e -> mayPointToList ctx e
+      | None -> []
+    in
+    RD.remove_vars_with new_rel local_vars; 
+    RD.remove_filter_with new_rel (fun var ->
+        match RV.to_cil_varinfo var with
+        | None -> false
+        | Some var -> 
+          ArrayMap.mem_varinfo var  || (*remove all ArrayMap vars*)
+          PointerMap.mem_varinfo var || (*remove all PointerMap vars*)
+          filterAllocVar var pointsToList (*remove all not reachable AllocSize vars*)
```

Comment:
If a block is allocated within a function, is the length information then forgotten after the return of that function?

Reply:
The length information is forgotten for array/pointer ghost variables as those are only local variables. However, alloc length ghost variables are not forgotten after a return, if the function returns a pointer that points to the alloc length ghost variable or if we assign the alloc length ghost variable to a global variable.